frontend/lib/framework/resolution-error-classes-list.cpp

```diff
@@ -516,22 +524,66 @@ void ErrorUseImportNotModule::write(ErrorWriterBase& wr) const {
   auto id = std::get<const ID>(info);
   auto moduleName = std::get<std::string>(info);
   auto useOrImport = std::get<const resolution::VisibilityStmtKind>(info);
+
   wr.heading(kind_, type_, id, "cannot '", useOrImport, "' '", moduleName,
-             "', which is not a module.");
+             "', which is not a ", allowedItem(useOrImport), ".");
   wr.message("In the following '", useOrImport, "' statement:");
   wr.code<ID, ID>(id, { id });
-  wr.message("Only modules and enums can be used with '", useOrImport,
-             "' statements.");
+  wr.message("Only ", allowedItems(useOrImport), " can be used with '",
+             useOrImport, "' statements.");
 }
 
 void ErrorUseImportUnknownMod::write(ErrorWriterBase& wr) const {
   auto id = std::get<const ID>(info);
   auto moduleName = std::get<std::string>(info);
   auto useOrImport = std::get<const resolution::VisibilityStmtKind>(info);
-  wr.heading(kind_, type_, id, "cannot find module '", moduleName,
-             "' for '", useOrImport, "'.");
+  auto& improperMatches = std::get<std::set<std::pair<ID, uast::asttags::AstTag>>>(info);
+
+  wr.heading(kind_, type_, id, "cannot find ", allowedItem(useOrImport),
+             " '", moduleName, "' for '", useOrImport, "' statement.");
   wr.message("In the following '", useOrImport, "' statement:");
   wr.code<ID, ID>(id, { id });
+  if (!improperMatches.empty()) {
+    wr.message("The following declarations are not covered by the '", useOrImport,
+               "' statement but seem similar to what you meant.");
+    // Do we need to tell the user how to use super.M or this.M?
+    bool explainRelativeImport = false;
+
+    for (auto idAndTag : improperMatches) {
+      // Separate out the various suggestions.
+      wr.message("");
+
+      // Use locationOnly to avoid printing 'in module M' for every suggestion.
```

Comment:
This is a good reminder that we should be able to tune the ErrorWriter not to display the header again for notes (when we do have it print headers).